orderer/consensus/etcdraft/chain.go

```diff
@@ -1235,6 +1238,37 @@ func pemToDER(pemBytes []byte, id uint64, certType string, logger *flogging.Fabr
 	return bl.Bytes, nil
 }
 
+// isConsensusUpgrade returns true if config block has a consensustype upgrade to smartbft.
+// in all other cases it returns false
+func (c *Chain) isConsensusUpgrade(block *common.Block) bool {
+	env := &common.Envelope{}
+	if err := proto.Unmarshal(block.Data.Data[0], env); err != nil {
+		return false
+	}
+
+	bundle, err := channelconfig.NewBundleFromEnvelope(env, c.CryptoProvider)
+	if err != nil {
+		return false
+	}
+
+	nextOrdererConfig, ok := bundle.OrdererConfig()
+	if ok {
+		if nextOrdererConfig.Capabilities().ConsensusTypeMigration() {
+			cfm := c.newConfigMetadata(block)
+			if cfm == nil {
+				return false
+			}
+			emptyMetaData := &etcdraft.ConfigMetadata{}
+			cfmBytes := protoutil.MarshalOrPanic(cfm)
+			emptyMetaDataBytes := protoutil.MarshalOrPanic(emptyMetaData)
+			if bytes.Equal(cfmBytes, emptyMetaDataBytes) && nextOrdererConfig.ConsensusType() == "smartbft" {
```

Comment:
You are checking here that the metadata is empty, but, this is post consensus. 

We need a check that ensures the validity of this config change when the config update is received *before* consensus.

Reply:
- Should we verify signatures on the config before consensus to ensure validity?
- What action is to be taken if validation fails?

Reply:
You already verify the signatures in the broadcast filter. 
If the validation fails you can reject the transaction.